app/webapp.py

```diff
@@ -25,11 +25,15 @@
     ThermostatMode,
     ThermostatSwing,
 )
+from aioswitcher.device.tools import convert_str_to_devicetype
```

Comment:
Based on the usage of this utility function and the modification to _openapi.yml_ spec, it looks like we're expecting users to pass the device type as strings in the query params:

```
Switcher Mini,
Switcher Power Plug,
Switcher Touch,
Switcher V2 (esp),
Switcher V2 (qualcomm),
Switcher V4, Switcher Breeze,
Switcher Runner,
Switcher Runner Mini,
Switcher Runner S11
```

Is that the right approach? We should support more url-friendly values and handle the translation ourselves.

Reply:
Why not? this is the same approach as [here](https://github.com/TomerFi/aioswitcher/blob/dev/src/aioswitcher/api/__init__.py#L638-L642)

Reply:
Python API vs URL Query Params.

For _Switcher V2 (qualcomm)_, the URL encoding will be _type=Switcher+V2+%28qualcomm%29_.
While something _type=v2qual_ will make the same sense and look much better.

Creating an enum, especially for this project, would be much better. We can easily convert the values before sending them to aioswitcher. Something like:

```
mini,
plug,
touch,
v2esp,
v2qual,
v4,
breeze,
runner,
runnermini,
runners11
```

Reply:
Done.

Reply:
Okay, I looked at the new implementation. Thank you—that is a step in the right direction, but we still need to do something else here.

Firstly, we don't need _aioswticher_'s _convert_str_to_devicetype_ utility function. You can remove this import.

Secondly, delete the added function _convert_url_devicetype_to_devicetype_; it's the only one that uses the abovementioned import. Instead of this function, add the following dictionary somewhere at the start of the file:
```python
DEVICES = {
    "mini": DeviceType.MINI,
    "plug": DeviceType.POWER_PLUG,
    "touch": DeviceType.TOUCH,
    "v2esp": DeviceType.V2_ESP,
    "v2qual": DeviceType.V2_QCA,
    "v4": DeviceType.V4,
    "breeze": DeviceType.BREEZE,
    "runner": DeviceType.RUNNER,
    "runnermini": DeviceType.RUNNER_MINI,
    "runners11": DeviceType.RUNNER_S11,
}
```

Next, modify all the invocations of the removed function from:
```python
device_type = convert_url_devicetype_to_devicetype(request.query[KEY_TYPE])
```
To:
```python
device_type = DEVICES[request.query[KEY_TYPE]]
```

This also means that if users pass an unknown key, i.e.:
```python
device_type = DEVICES["wrongdevicekey"]
```
 An error will be thrown. Which is a good thing! Just make sure you're aware of that when modifying the code.

Reply:
Done.